### PR TITLE
Add periodic contributor update workflow

### PR DESCRIPTION
**Pull request by arturict**
## Summary

- schedule a monthly contributor-list maintenance reminder
- create one open GitHub issue instead of committing generated changes as a bot
- keep the actual update and commit attributable to the maintainer who performs it
- avoid duplicate reminders while an existing `Update contributor list` issue remains open
- use only `issues: write`; do not create branches, pull requests, or email notifications

## Reminder workflow

The issue includes the existing maintenance command:

```console
python3 contrib/lint/analyze-contributors.py --update-file CONTRIBUTORS.md
```

A maintainer can run it, review the generated diff, and commit it under their own identity. Manual workflow dispatch follows the same duplicate-safe path as the monthly schedule.

## Validation

- `actionlint` v1.7.7 passes
- the live read-only `gh issue list` query returns the expected numeric count
- permissions assertion confirms `issues: write` with no contents or pull-request access
- `git diff --check` passes

Related to #912.

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 solvcon is made possible by the following contributors.
 
+To update this list, check out `master` with full Git history and run:
```
Comment (tigercosmos) [inline CONTRIBUTORS.md]: I don't think we need these lines in CONTRIBUTORS.md

File: .github/workflows/update_contributors.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Update Contributors
+
+on:
+  schedule:
+    - cron: '17 8 1 * *'
```
Comment (tigercosmos) [inline .github/workflows/update_contributors.yml:6]: Maybe add a comment to mention it's monthly?

File: .github/workflows/update_contributors.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  workflow_dispatch:
+
+concurrency:
+  group: update-contributors
+  cancel-in-progress: true
```
Comment (tigercosmos) [inline .github/workflows/update_contributors.yml]: Change to `false` to ensure completion.

File: .github/workflows/update_contributors.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          fetch-depth: 0
+
+      - name: Update contributor list
+        run: |
+          python3 contrib/lint/analyze-contributors.py \
+            --update-file CONTRIBUTORS.md
+
+      - name: Create contributor update pull request
+        id: contributor_pr
+        uses: peter-evans/create-pull-request@v8
```
Comment (tigercosmos) [inline .github/workflows/update_contributors.yml]: I would prefer to use the native GitHub CLI in this case for security issues.
```
      - name: Create Pull Request
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          gh pr create \
            --title "Automated PR from ${{ github.ref_name }}" \
            --body "This PR was automatically created by GitHub Actions." \
            --base main \
            --head ${{ github.ref_name }}
```

File: CONTRIBUTORS.md (hunk 1 above)
Comment (arturict) [reply CONTRIBUTORS.md]: Removed the added instructions from `CONTRIBUTORS.md`; the generated file is back to its existing content format.

File: .github/workflows/update_contributors.yml (hunk 2 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml:6]: Added a comment documenting that the cron runs monthly on the first day at 08:17 UTC.

File: .github/workflows/update_contributors.yml (hunk 3 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Changed `cancel-in-progress` to `false` so an in-flight contributor update completes.

File: .github/workflows/update_contributors.yml (hunk 4 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Replaced `peter-evans/create-pull-request` with native `git` and `gh`. The workflow updates a stable automation branch with `--force-with-lease`, reuses an existing open PR, and skips duplicate commits/notifications when that branch already contains the generated file.

**Comment by arturict** [comment]
@tigercosmos Thanks for the review. I addressed all four points in c39ec8d: removed the added CONTRIBUTORS.md instructions, documented the monthly cron, set cancel-in-progress to false, and replaced peter-evans/create-pull-request with native git/gh while preserving stable-branch idempotence. I replied in and resolved each thread. The updated Linux/macOS lint jobs and Ubuntu/macOS/Windows release builds are all green.

File: .github/workflows/update_contributors.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            git fetch origin "+$branch:refs/remotes/origin/$branch"
+          fi
+          if git rev-parse --verify "origin/$branch" >/dev/null 2>&1 && \
+              git diff --quiet "origin/$branch" -- CONTRIBUTORS.md; then
+            echo "changed=false" >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (tigercosmos) [inline .github/workflows/update_contributors.yml]: Not sure if this is correct.

File: .github/workflows/update_contributors.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            --state open \
+            --json url \
+            --jq '.[0].url')
+
+          if [ -z "$pr_url" ]; then
+            pr_url=$(gh pr create \
+              --base master \
+              --head "$branch" \
+              --title "Update contributor list" \
+              --body $'This pull request updates the contributor list from the commit history and source-file copyright headers. Please review the generated changes before merging.\n\nRelated to #912.')
```
Comment (tigercosmos) [inline .github/workflows/update_contributors.yml]: Nitpick: remove `Related to #921`.

**Review by tigercosmos — COMMENTED** [review]
@arturict  Thanks for your quick update. Overall it looks good to me. Could you set up the CI on your fork and trigger this workflow to ensure it works correctly?

File: .github/workflows/update_contributors.yml (hunk 5 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Good question. The fork run exposed a retry edge case: the branch push could succeed while PR creation failed, and the next run would previously stop on the matching branch content. I now query the open PR first and only skip when both the generated file matches and an open PR exists. The controlled change run created the PR successfully, and a second run reused it without another commit: https://github.com/arturict/solvcon/actions/runs/29204770260 and https://github.com/arturict/solvcon/actions/runs/29204784367

File: .github/workflows/update_contributors.yml (hunk 6 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Removed the issue reference from the generated pull request body.

**Comment by arturict** [comment]
Thanks. I set up the workflow on my fork and exercised both paths.

- No-change dispatch succeeded as expected, with no PR and no email: https://github.com/arturict/solvcon/actions/runs/29204677096
- A controlled fork-only contributor change exercised the full native git/gh path and created a review PR successfully: https://github.com/arturict/solvcon/actions/runs/29204770260
- A second dispatch confirmed idempotence by reusing the existing PR without another update: https://github.com/arturict/solvcon/actions/runs/29204784367

The email step is guarded to run only in `solvcon/solvcon`, so fork validation could not send mail to the project list. The test PR and automation branch were removed afterward, and the temporary contributor change was reverted. The test also exposed and helped fix the retry edge case described in the inline thread.

File: .github/workflows/update_contributors.yml (hunk 5 above)
Comment (tigercosmos) [reply .github/workflows/update_contributors.yml]: I see. 41898282 is the standard number of the bot.

**Review by tigercosmos — APPROVED** [review]
I saw it successfully triggered a PR https://github.com/arturict/solvcon/pull/1. LGTM

File: .github/workflows/update_contributors.yml (hunk 5 above)
Comment (yungyuc) [reply .github/workflows/update_contributors.yml]: The user name is a show stopper.  The automatic PR creation uses a bot, and in the commit history there will not be information for who authorizes the contributor update.  Too little accountability.  We should not use a bot PR for the contributor list.

Instead, we should create an issue to remind the maintainers to update the contributor list.  It allows to use more strict permission, avoid creating a branch in this upstream repository and sending an email.

File: .github/workflows/update_contributors.yml [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    - cron: '17 8 1 * *'
+  workflow_dispatch:
+
+concurrency:
+  group: update-contributors
+  cancel-in-progress: false
+
+permissions:
+  contents: write
+  pull-requests: write
```
Comment (yungyuc) [inline .github/workflows/update_contributors.yml]: By creating an issue, we will not need `contents: write` and will not create branches in this upstream repository:

```suggestion
permissions:
  issue: write
```

**Review by yungyuc — CHANGES_REQUESTED** [review]
> Thanks. I set up the workflow on my fork and exercised both paths.

Solid practice. It saves a lot of back-n-force discussions. Thanks.

Point to address:
- [x] Create an issue instead of a PR.

@ExplorerRay please take a look.

----

BTW, no need to resolve comments.  It is more effective to leave a word to summarize how you address under them.

File: .github/workflows/update_contributors.yml (hunk 5 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Agreed. I replaced the bot branch/commit/PR path with a monthly reminder issue in 8bd98d9. The workflow now leaves the analyzer run, diff review, and commit to an identifiable maintainer. It also removes the email step and skips creation when an exact-title reminder is already open. `actionlint` passes.

File: .github/workflows/update_contributors.yml (hunk 7 above)
Comment (arturict) [reply .github/workflows/update_contributors.yml]: Changed to the GitHub Actions permission name `issues: write` in 8bd98d9. `contents` and `pull-requests` access are both removed, and the workflow no longer checks out or pushes repository content.

File: .github/workflows/update_contributors.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+concurrency:
+  group: update-contributors
+  cancel-in-progress: false
+
+permissions:
+  issues: write
+
+jobs:
+  remind:
+    runs-on: ubuntu-latest
```
Comment (yungyuc) [inline .github/workflows/update_contributors.yml:18]: The project prefers to use explicit machine for CI, but it is fine to use `-latest` for this workflow.

File: .github/workflows/update_contributors.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
+name: Update Contributors
```
Comment (yungyuc) [inline .github/workflows/update_contributors.yml:1]: It will be more accurate to name it like review contributors, but update contributors is OK for now.

File: .github/workflows/update_contributors.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+jobs:
+  remind:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Create contributor update reminder
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          title='Update contributor list'
```
Comment (yungyuc) [inline .github/workflows/update_contributors.yml:25]: It is right to use "update" in the issue title.  It will be better to include date in the title will help organize the issue, but it is OK to leave it as is for now.

**Review by yungyuc — APPROVED** [review]
LGTM

I am merging.  @ExplorerRay @chestercheng please comment.

File: .github/workflows/update_contributors.yml (hunk 10 above)
Comment (chestercheng) [reply .github/workflows/update_contributors.yml:25]: I think it would be useful to add a dedicated label to this so the workflow can identify it more reliably instead of relying only on the title.

File: .github/workflows/update_contributors.yml [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+          open_reminders=$(gh issue list \
+            --repo "$GITHUB_REPOSITORY" \
+            --state open \
+            --limit 100 \
+            --json title \
+            --jq "map(select(.title == \"$title\")) | length")
+
+          if [ "$open_reminders" -gt 0 ]; then
+            echo 'An open contributor update reminder already exists.'
+            exit 0
```
Comment (chestercheng) [inline .github/workflows/update_contributors.yml:35]: If the issue is still open when the workflow runs again, maybe it could post a reminder comment instead of doing nothing. That would help keep the reminder visible.

File: .github/workflows/update_contributors.yml (hunk 10 above)
Comment (yungyuc) [reply .github/workflows/update_contributors.yml:25]: A dedicated label for a monthly review run sounds like an overkill.

File: .github/workflows/update_contributors.yml (hunk 11 above)
Comment (yungyuc) [reply .github/workflows/update_contributors.yml:35]: Agreed.
